### ISSUE TITLE
How to use gpu mode

### ISSUE THREAD

**Issue by DavieHR** [DmitryUlyanov/texture_nets#8]
well, i run the pretrain model code , and i encounter the problem:attempt to index field 'gradInput' (a nil value).And i think i use the gpu, and what's wrong?

**Comment by DmitryUlyanov** [comment]
Looks like pretrained models doesn't work even on CPU now, there was a bug in network structure, and it worked because there was no exception in Batchnorm layer if incorrect parameters passed. They've changed it now. I'll try to generate new models and put them here.

**Comment by DavieHR** [comment]
OK Thx

**Comment by DmitryUlyanov** [comment]
I found that models work on GPU, if you define 

function nn.BatchNormalization:checkInputDim(input)
end

somewhere. It will probably fail on CPU, but good as temporary fix.
